### PR TITLE
[gettext] Add feature tools

### LINKED ISSUE #13518: [gettext] msgfmt (and other gettext bin tools) missing on macOS & Windows
**Describe the bug**
macOS nor Windows  come with gettext by default. There is a gettext port in vcpkg,[ but it doesn't ship the \bin tools](https://github.com/microsoft/vcpkg/pull/245#issue-92014084). The `vcpkg_find_acquire_program()` does not offer it, either. Because of that reason, ports that require msgfmt binary tools (e.g. `kf5i18n`) fail to build on macOS and Windows.

~~What's even more puzzling, though, is that the vcpkg OS X Azure Pipeline [builds it just fine](https://dev.azure.com/vcpkg/public/_build/results?buildId=42873&view=logs&j=a83b7983-e539-5449-9e09-337ff5f52283&t=8636fc5b-b802-51ab-60e0-76e11f68a179&l=800), despite that gettext [is not one of the home-brew prerequisites](https://github.com/microsoft/vcpkg/blob/master/scripts/azure-pipelines/osx/configuration/VagrantFile) installed by vagrant.~~ EDIT: these pipeline builds were in fact false-positives and [were never actually executed](https://github.com/microsoft/vcpkg/pull/13467#issuecomment-692119190).

**Environment**
- OS: macOS 10.15.6, as well as Windows 10
- Compiler: Apple clang version 11.0.3 (clang-1103.0.32.62)

**To Reproduce**
1. Make sure you do not have a gettext brew package installed
2. `vcpkg install kf5i18`
3. See the error.

**Expected behavior**
1. Azure Pipeline build should fail just like the local build does
2. `vcpkg_find_acquire_program()` should include gettext tools

~~Alternatively, the way that OS X Azure Pipeline delivers a functioning set of gettext tools should be explicitly documented.~~ as explained above, this was a false-positive.

**Failure logs** 
```
CMake Error at /Volumes/External/Sourcecode/vcpkg/downloads/tools/cmake-3.17.2-osx/cmake-3.17.2-Darwin-x86_64/CMake.app/Contents/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Gettext (missing: GETTEXT_MSGMERGE_EXECUTABLE
  GETTEXT_MSGFMT_EXECUTABLE)
Call Stack (most recent call first):
  /Volumes/External/Sourcecode/vcpkg/downloads/tools/cmake-3.17.2-osx/cmake-3.17.2-Darwin-x86_64/CMake.app/Contents/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  /Volumes/External/Sourcecode/vcpkg/downloads/tools/cmake-3.17.2-osx/cmake-3.17.2-Darwin-x86_64/CMake.app/Contents/share/cmake-3.17/Modules/FindGettext.cmake:81 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  /Volumes/External/Sourcecode/vcpkg/scripts/buildsystems/vcpkg.cmake:478 (_find_package)
  /Volumes/External/Sourcecode/vcpkg/buildtrees/kf5i18n/x64-osx-dbg/cmake/KF5I18nMacros.cmake:5 (find_package)
  CMakeLists.txt:66 (include)
```

### PR DESCRIPTION
**Pull request by JackBoosY**
Fixes #13518.

### CODE REVIEW

**Comment by Neumann-A** [comment]
there are two scripts which use absolute paths somewhere arround `gettext-tools/misc/*.in` these need to be patched to calculated the prefix given the script installed path instead.

**Comment by autoantwort** [comment]
> @Neumann-A I noticed that libtools didn't generated, what's wrong?

On my mac the follwing tools get generated: autopoint gettext msgattrib msgcat msgcmp msgcomm msgconv msgen msgexec msgfilter msgfmt msggrep msginit msgmerge msgunfmt msguniq ngettext recode-sr-latin xgettext gettextize envsubst  
Is there something missing?

File: ports/gettext/portfile.cmake [hunk 1]
```diff
@@ -43,23 +43,40 @@ if(VCPKG_TARGET_IS_WINDOWS)
...
+        vcpkg_install_make()
+    else()
+        vcpkg_install_make(SUBPATH "/intl")
+    endif()
 endif()
 
-
 # Handle copyright
 file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/gettext)
 file(RENAME ${CURRENT_PACKAGES_DIR}/share/gettext/COPYING ${CURRENT_PACKAGES_DIR}/share/gettext/copyright)
```
Comment (autoantwort) [inline ports/gettext/portfile.cmake:82]:
```suggestion
file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

**Comment by Neumann-A** [comment]
@JackBoosY libtools is not part of gettext? autopoint is. (which is just a script)

`autopoint.in` and `gettextize.in` needs to be changed to have:
```
# Set variables
# - gettext_datadir     directory where the data files are stored.
prefix="$(dirname "$0")/../../../"
datarootdir="${prefix}/share/gettext"
```
to be relocatable. Otherwise they hardcode the absolute path

**Pull request closed without merging**